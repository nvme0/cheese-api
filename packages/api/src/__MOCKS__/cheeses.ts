import { Cheese } from "../types";

export const organicFeta: Cheese = {
  id: "9ac773e1-c953-4ba8-90d3-02e0eeddef44",
  name: "Organic Feta",
  imageUrl: "https://cheese.com/media/img/cheese/Feta_.jpg",
  color: "white",
  pricePerKilo: 549,
};

export const somersetBrie: Cheese = {
  id: "3ac413d1-36e5-43fd-a050-785a522cd940",
  name: "Somerset Brie",
  imageUrl: "https://cheese.com/media/img/cheese/brie.jpg",
  color: "white",
  pricePerKilo: 325,
};

export const mascarpone: Cheese = {
  id: "c56777da-0d78-4bad-ad35-57c546ce6387",
  name: "Mascarpone (Australian)",
  imageUrl: "https://cheese.com/media/img/cheese/1186-mascarpone_australian.jpg",
  color: "ivory",
  pricePerKilo: 199,
};

export const cottageCheese: Cheese = {
  id: "a2553e96-05e7-48bd-ab8b-8c70350be2f0",
  name: "Cottage Cheese (Australian)",
  imageUrl: "https://cheese.com/media/img/cheese/Cottagecheese.jpg",
  color: "white",
  pricePerKilo: 399,
};

export const dryJack: Cheese = {
  id: "f411ab0a-306e-47e0-a7ef-31c4b4ede802",
  name: "Dry Jack",
  imageUrl: "https://cheese.com/media/img/cheese/427px-Sonoma_Dry_Jack.jpg",
  color: "pale yellow",
  pricePerKilo: 249,
};

export default [organicFeta, somersetBrie, mascarpone, cottageCheese, dryJack];
