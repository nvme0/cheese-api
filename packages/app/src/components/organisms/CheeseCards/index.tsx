import React from "react";
import { Box } from "@material-ui/core";

import useStyles from "./styles";
import { Cheese } from "@app/types";
import CheeseCard from "@molecules/CheeseCard";

export interface Props {
  cheeses: Cheese[];
}

const CheeseCards = (props: Props) => {
  const { cheeses } = props;
  const classes = useStyles();

  return (
    <Box className={classes.root}>
      {cheeses.map(({ id, name, color, imageUrl, pricePerKilo }) => (
        <CheeseCard key={id} title={name} color={color} imageUrl={imageUrl} pricePerKilo={pricePerKilo} />
      ))}
    </Box>
  );
};

export default CheeseCards;
