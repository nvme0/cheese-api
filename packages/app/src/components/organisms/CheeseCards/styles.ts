import { makeStyles, useTheme } from "@material-ui/core/styles";

export const appBarHeight = 58;
export const footerHeight = 48;

const styles = makeStyles((theme) => ({
  root: {
    display: "grid",
    gridTemplateColumns: "1fr 1fr 1fr",
    gridGap: theme.spacing(2),
    maxWidth: theme.breakpoints.values.lg,
    margin: `${theme.spacing(2)}px auto`,
    padding: `0 ${theme.spacing(2)}px`,
    [theme.breakpoints.down("sm")]: {
      gridTemplateColumns: "1fr 1fr",
      margin: `${theme.spacing(2)}px 0`,
      padding: `0 ${theme.spacing(1)}px`,
    },
    [theme.breakpoints.down("xs")]: {
      gridTemplateColumns: "1fr",
    },
  },
}));

const useStyles = () => {
  const theme = useTheme();
  return styles(theme);
};

export default useStyles;
