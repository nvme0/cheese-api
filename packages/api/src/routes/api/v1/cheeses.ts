import { Router } from "express";
import cheeses from "../../../__MOCKS__/cheeses";

const router = Router();

router.get("/", (req, res) => {
  res.send(cheeses);
});

router.get("/:id", (req, res) => {
  const {
    params: { id },
  } = req;

  const requestedCheese = cheeses.find((cheese) => cheese.id === id);
  if (requestedCheese) {
    res.send(requestedCheese);
  } else {
    res.sendStatus(404);
  }
});

export default router;
