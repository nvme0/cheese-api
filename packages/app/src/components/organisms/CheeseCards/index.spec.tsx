import React from "react";
import { render } from "@testing-library/react";

import CheeseCards from ".";
import cheeses from "../../../__MOCKS__/cheeses";

describe("CheeseCards", () => {
  it("renders correctly", () => {
    const component = render(<CheeseCards cheeses={cheeses} />);
    expect(component).toMatchSnapshot();
  });
});
