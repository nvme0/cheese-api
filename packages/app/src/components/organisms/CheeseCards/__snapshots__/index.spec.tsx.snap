// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheeseCards renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-2 makeStyles-root-1"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-3"
              style="background-image: url(https://cheese.com/media/img/cheese/Feta_.jpg);"
              title="Organic Feta"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Organic Feta
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                />
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Color: 
                  white
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Price Per Kilo: $
                  5.49
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-3"
              style="background-image: url(https://cheese.com/media/img/cheese/brie.jpg);"
              title="Somerset Brie"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Somerset Brie
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                />
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Color: 
                  white
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Price Per Kilo: $
                  3.25
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-3"
              style="background-image: url(https://cheese.com/media/img/cheese/1186-mascarpone_australian.jpg);"
              title="Mascarpone (Australian)"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Mascarpone (Australian)
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                />
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Color: 
                  ivory
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Price Per Kilo: $
                  1.99
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-3"
              style="background-image: url(https://cheese.com/media/img/cheese/Cottagecheese.jpg);"
              title="Cottage Cheese (Australian)"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Cottage Cheese (Australian)
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                />
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Color: 
                  white
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Price Per Kilo: $
                  3.99
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-3"
              style="background-image: url(https://cheese.com/media/img/cheese/427px-Sonoma_Dry_Jack.jpg);"
              title="Dry Jack"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Dry Jack
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                />
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Color: 
                  pale yellow
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Price Per Kilo: $
                  2.49
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-2 makeStyles-root-1"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-3"
            style="background-image: url(https://cheese.com/media/img/cheese/Feta_.jpg);"
            title="Organic Feta"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Organic Feta
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                white
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                5.49
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-3"
            style="background-image: url(https://cheese.com/media/img/cheese/brie.jpg);"
            title="Somerset Brie"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Somerset Brie
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                white
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                3.25
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-3"
            style="background-image: url(https://cheese.com/media/img/cheese/1186-mascarpone_australian.jpg);"
            title="Mascarpone (Australian)"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Mascarpone (Australian)
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                ivory
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                1.99
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-3"
            style="background-image: url(https://cheese.com/media/img/cheese/Cottagecheese.jpg);"
            title="Cottage Cheese (Australian)"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Cottage Cheese (Australian)
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                white
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                3.99
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-3"
            style="background-image: url(https://cheese.com/media/img/cheese/427px-Sonoma_Dry_Jack.jpg);"
            title="Dry Jack"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Dry Jack
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                pale yellow
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                2.49
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
